### PR TITLE
A few terminology changes in module names of the universe API

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** code documentation

This is a mini-change to the universe API so as to make some terms easier to grasp. It does:
- `Univ.AUContext` -> `Univ.AbstractContext`
- `LSet` -> `Level.Set`, `LMap` -> `Level.Map`
- `Univ.Constraint` -> `Univ.Constraints`

See also Zulip's [discussion](https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs.20.26.20plugin.20devs/topic/terminology.20used.20for.20universe.20API).

- [X] `dev/doc/changes.md` updated

### CODE REVIEW

File: dev/doc/changes.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+## Changes between Coq 8.14 and Coq 8.15
+
+### Terminology used in the universes API
+
+- Renaming `Univ.Constraint` into `Univ.Constraints`
+
+- Renaming `LSet` into `LevelSet` and `LMap` into `LevelMap`
```
Comment (SkySkimmer) [inline dev/doc/changes.md:7]:
```suggestion
- Renaming `LSet` into `Level.Set` and `LMap` into `Level.Map`
```

**Review by SkySkimmer — APPROVED** [review]

File: dev/doc/changes.md (hunk 1 above)
Comment (herbelin) [reply dev/doc/changes.md]: Thanks

**Comment by Alizter** [comment]
I believe the ci failure for bedrock2 is unrelated. But the following need overlays:
* coq_dpdgraph
* elpi
* equations
* mtac2

The test-suite also needs to be updated.

**Comment by herbelin** [comment]
Should be ok now without overlays (until the end of deprecation phase). Bedrock2 still unrelated.

**Comment by SkySkimmer** [comment]
The rebase label haunts this PR

**Comment by herbelin** [comment]
Don't worry, I'm quickly rebasing...

**Comment by SkySkimmer** [comment]
Elpi needs overlay.

**Comment by herbelin** [comment]
> Elpi needs overlay.

It was a previous overlay (#14692) still not taken into account at the time of the bench. I rerun and it is now ok.

**Comment by SkySkimmer** [comment]
@coqbot: merge now

**Comment by herbelin** [comment]
Thanks!

**Pull request merged**